packages/kbn-performance-testing-dataset-extractor/src/es_client.ts

```diff
@@ -88,81 +100,92 @@ const addRangeFilter = (range: { startTime: string; endTime: string }): QueryDsl
   };
 };
 
-export function initClient(options: ClientOptions, log: ToolingLog) {
-  const client = new Client({
-    node: options.node,
-    auth: {
-      username: options.username,
-      password: options.password,
-    },
-  });
+export class ESClient {
+  client: Client;
+  log: ToolingLog;
 
-  return {
-    async getKibanaServerTransactions(
-      buildId: string,
-      journeyName: string,
-      range?: { startTime: string; endTime: string }
-    ) {
-      const filters = [
-        { field: 'transaction.type', value: 'request' },
-        { field: 'processor.event', value: 'transaction' },
-        { field: 'labels.testBuildId', value: buildId },
-        { field: 'labels.journeyName', value: journeyName },
-      ];
-      const queryFilters = filters.map((filter) => addBooleanFilter(filter));
-      if (range) {
-        queryFilters.push(addRangeFilter(range));
-      }
-      return await this.getTransactions(queryFilters);
-    },
-    async getFtrTransactions(buildId: string, journeyName: string) {
-      const filters = [
-        { field: 'service.name', value: 'functional test runner' },
-        { field: 'processor.event', value: 'transaction' },
-        { field: 'labels.testBuildId', value: buildId },
-        { field: 'labels.journeyName', value: journeyName },
-        { field: 'labels.performancePhase', value: 'TEST' },
-      ];
-      const queryFilters = filters.map((filter) => addBooleanFilter(filter));
-      return await this.getTransactions(queryFilters);
-    },
+  constructor(options: ClientOptions, log: ToolingLog) {
+    this.client = new Client({
+      node: options.node,
+      auth: {
+        username: options.username,
+        password: options.password,
+      },
+    });
+    this.log = log;
+  }
 
-    async getTransactions(queryFilters: QueryDslQueryContainer[]) {
-      const searchRequest: SearchRequest = {
-        body: {
-          track_total_hits: true,
-          sort: [
-            {
-              '@timestamp': {
-                order: 'asc',
-                unmapped_type: 'boolean',
-              },
+  async getTransactions<T>(queryFilters: QueryDslQueryContainer[]) {
+    const searchRequest: SearchRequest = {
+      body: {
+        track_total_hits: true,
+        sort: [
+          {
+            '@timestamp': {
+              order: 'asc',
+              unmapped_type: 'boolean',
             },
-          ],
-          size: 10000,
-          stored_fields: ['*'],
-          _source: true,
-          query: {
-            bool: {
-              must: [],
-              filter: [
-                {
-                  bool: {
-                    filter: queryFilters,
-                  },
+          },
+        ],
+        size: 10000,
+        stored_fields: ['*'],
```

Comment:
Assuming we just want to retrieve the entire document, we can get rid of the `stored_fields` and the `_source` parameter.